src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.test.ts

```diff
@@ -0,0 +1,307 @@
+import {
+  GoogleAnalysisStorageContract,
+  GoogleStorage,
+  GoogleStorageContract
+} from 'src/libs/ajax/GoogleStorage'
+import { CloudProviderType, cloudProviderTypes } from 'src/libs/workspace-utils'
+import { defaultAzureWorkspace, defaultGoogleWorkspace } from 'src/pages/workspaces/workspace/analysis/_testData/testData'
+import {
+  AbsolutePath, AnalysisFile, getDisplayName, getExtension, getFileName, useAnalysisFiles
+} from 'src/pages/workspaces/workspace/analysis/file-utils'
+import {
+  AnalysisDuplicator,
+  AnalysisDuplicatorProps
+} from 'src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator'
+import {
+  getToolFromFileExtension, ToolLabel,
+  toolLabelTypes,
+  tools
+} from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { asMockedFn } from 'src/testing/test-utils'
+
+
+jest.mock('src/components/Modal', () => {
+  const { div, h } = jest.requireActual('react-hyperscript-helpers')
+  const originalModule = jest.requireActual('src/components/Modal')
+  return {
+    ...originalModule,
+    __esModule: true,
+    default: props => div({ id: 'modal-root' }, [
+      h(originalModule.default, { onAfterOpen: jest.fn(), ...props })
+    ])
+  }
+})
+
+jest.mock('src/libs/notifications', () => ({
+  notify: (...args) => {
+    console.debug('######################### notify')/* eslint-disable-line */
+    console.debug({ method: 'notify', args: [...args] })/* eslint-disable-line */
+  }
+}))
+
+jest.mock('src/libs/ajax/GoogleStorage')
+jest.mock('src/libs/ajax/AzureStorage')
+
+jest.mock('src/pages/workspaces/workspace/analysis/file-utils', () => {
+  const originalModule = jest.requireActual('src/pages/workspaces/workspace/analysis/file-utils')
+  return {
+    ...originalModule,
+    useAnalysisFiles: jest.fn()
+  }
+})
+
+const getTestFile = (abs: AbsolutePath, cloudProvider: CloudProviderType = cloudProviderTypes.GCP): AnalysisFile => ({
+  name: abs,
+  ext: getExtension(abs),
+  displayName: getDisplayName(abs),
+  fileName: getFileName(abs),
+  tool: getToolFromFileExtension(getExtension(abs)) as ToolLabel,
+  lastModified: new Date().getTime(),
+  cloudProvider
+})
+
+const baseTestFile: AnalysisFile = getTestFile('test/file0.ipynb' as AbsolutePath)
+
+const onDismiss = jest.fn()
+const onSuccess = jest.fn()
+
+const defaultModalProps: AnalysisDuplicatorProps = {
+  destroyOld: false,
+  fromLauncher: false,
+  printName: baseTestFile.fileName,
+  toolLabel: toolLabelTypes[tools.Jupyter.label],
+  workspaceInfo: defaultGoogleWorkspace.workspace,
+  onDismiss, onSuccess
+}
+
+describe('AnalysisDuplicator', () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+    // Arrange
+    const googleStorageMock: Partial<GoogleStorageContract> = ({
+      listAnalyses: jest.fn(() => Promise.resolve([]))
+    })
+    asMockedFn(GoogleStorage).mockImplementation(() => googleStorageMock as GoogleStorageContract)
```

Comment:
Since this test mocks `useAnalysisFiles`, do `listAnalyses` and `listNotebooks` need to be mocked?

Reply:
hmm, not anymore it seems